File: frontend/src/services/BaseUserService.js
import axios from "axios";

export default class BaseUserService{
    getAll() {
        return axios.get("http://localhost:8080/api/baseUser/getAll")
    }

    getById(id) {
        return axios.get(`http://localhost:8080/api/baseUser/getById?id=${id}`)
    }

    getByPhoneNumber(phoneNumber) {
        return axios.get(`http://localhost:8080/api/baseUser/getByPhoneNumber?phoneNumber=${phoneNumber}`)
    }

    getByEmail(email) {
        return axios.get(`http://localhost:8080/api/baseUser/getByEmail?email=ad${email}`)
    }
}

File: backend/src/main/java/ecrow/backend/business/abstracts/BaseUserService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.core.utilities.results.Result;
import ecrow.backend.entities.concretes.BaseUser;
import ecrow.backend.entities.dtos.WithdrawBalanceDto;

import java.util.List;

public interface BaseUserService {
    DataResult<List<BaseUser>> getAll();
    DataResult<BaseUser> getById(Integer id);
    DataResult<BaseUser> getByEmail(String email);
    DataResult<BaseUser> getByPhoneNumber(String phoneNumber);
    Result withdrawBalance(WithdrawBalanceDto withdrawBalanceDto);
}
